io/disk/tiobench.py

```diff
@@ -0,0 +1,96 @@
+
+# Based on code by
+#   Author: Lucas Meneghel Rodrigues <lmr@redhat.com>
+#   Author: Xu Tian <xutian@redhat.com>
+#   Author: Jeff Moyer <jmoyer@redhat.com>
+#   Author: Dmitry Monakhov <dmonakhov@openvz.org>
+#   Copyright: 2008 - 2015 Red Hat, Inc.
+#   Copyright: 2013 Red Hat, Inc.
+#   Copyright: 2010 Red Hat, Inc.
+#   Copyright: 2012 <dmonakhov@openvz.org>
+#   https://github.com/autotest/autotest-client-tests/tree/master/tiobench
+
+
+import os
+
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import build
+from avocado.utils import git
+from avocado.utils import genio
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Tiobench(Test):
+
+    """
+    Avocado module for tiobench.
+    tiobench is a multi-threaded I/O benchmark.
+    It is used to measure file system performance in
+    four basic operations: sequential read, random read,
+    sequential write, and random write.
+
+    """
+
+    def setUp(self):
+        """
+        Build tiobench.
+        Source:
+        https://github.com/mkuoppal/tiobench.git
+
+        """
+        s_mngr = SoftwareManager()
+        if not s_mngr.check_installed("gcc") and not s_mngr.install("gcc"):
+            self.error('Gcc is needed for the test to be run')
+        git.get_repo('https://github.com/mkuoppal/tiobench.git',
+                     destination_dir=self.srcdir)
+        os.chdir(self.srcdir)
+        build.make(self.srcdir)
+
+    def test(self):
+        """
+        Runs two iterations of test.
+        :params target: The directory in which to test.
+                        Defaults to ., the current directory.
+        :params blocks: The blocksize in Bytes to use. Defaults to 4096.
+        :params threads: The number of concurrent test threads.
+        :params size: The total size in MBytes of the files may use together.
+        :params num_runs: This number specifies over how many runs
+                          each test should be averaged.
+        :params iterations: Number of iterations the test needs to be run.
+        """
+        target = self.params.get('target', default='/mnt')
+        blocks = self.params.get('blocks', default=4096)
+        threads = self.params.get('threads', default=10)
+        size = self.params.get('size', default=1024)
+        num_runs = self.params.get('numruns', default=2)
+        iterations = self.params.get('iterations', default=2)
+        results = process.system_output('./tiobench.pl --target {} --block={}'
+                                        ' --threads={} --size={} --numruns={}'
+                                        .format(target, blocks, threads,
+                                                size, num_runs))
+        self.log.info(results)
+        for iteration in range(1, iterations + 1):
+            results_path = os.path.join(self.outputdir,
+                                        'raw_output_%s' % iteration)
+
```

Comment:
Basically the multiplexer is way to run combinations. Right now there is no way to just provide a number of iterations, so you'd have to create the variants similarly to `examples/tests/doublefree_nasty.py.data/iterations.yaml`. But we plan to extend the multiplexer, currently it's about the internal filters, but then we can focus on specifying arguments by lists, or even range as those could be useful.